### ISSUE TITLE
Source tarball for 4.0.0-beta.1 contains stale libtransmission/version.h

### ISSUE THREAD

**Issue by floppym** [transmission/transmission#3901]
The [source tarball](https://github.com/transmission/transmission/releases/download/4.0.0-beta.1/transmission-4.0.0-beta.1+r98cf7d9b3c.tar.xz) for 4.0.0-beta.1 contains a stale copy of libtransmission/version.h.
```
transmission-4.0.0-beta.1+r98cf7d9b3c % cat libtransmission/version.h
#pragma once

#define PEERID_PREFIX             "-TR3AAZ-"
#define USERAGENT_PREFIX          "3.10.10-dev"
#define VCS_REVISION              "1a42769e41"
#define VCS_REVISION_NUM          1a42769e41
#define SHORT_VERSION_STRING      "3.10.10-dev"
#define LONG_VERSION_STRING       "3.10.10-dev (1a42769e41)"
#define VERSION_STRING_INFOPLIST  3.10.10-dev
#define BUILD_STRING_INFOPLIST    14714.3.10.10
#define MAJOR_VERSION             3
#define MINOR_VERSION             10
#define PATCH_VERSION             10
#define TR_NIGHTLY_RELEASE        1
```
Running cmake in a separate build directory will generate another copy of version.h with the correct version number.

However, this will cause some components to be built with the correct version (4.0.0-beta.1), and some components to be built with version 3.10.10-dev, depending on the order of the include directories given on the compiler command line for a given component.

In my own testing, it appears that libtransmission.a gets built with the stale version.h in the source directory, and all other components get build with the fresh copy in the build directory.

**Comment by ckerr** [comment]
Ugh

**Comment by ckerr** [comment]
This is my bad. :disappointed: 

I built the tarball from the same checkout that I'd done some other work in, including testing of a script that generates `version.h`. It's not supposed to be in the source directory at all; it's generated in the build directory by CMake. Looks like what happens is if someone _does_ leave a copy in there, _then_ it gets picked up by CPack.

tldr, always build tarballs from a fresh checkout.

I'm going to close this ticket since it's too late to fix this for beta.1 and it's not something in the code or CMake that needs to get "fixed" for future releases -- but I'll be sure to use fresh checkouts from now on. Thanks for reporting this @floppym
